### ISSUE TITLE
eleventyComputed property with pagination is computed with empty data

### ISSUE THREAD

**Issue by souserge** [11ty/eleventy#2837]
### Operating system

Fedora Linux 37 (Workstation edition)

### Eleventy

2.0.0

### Describe the bug

I stumbled across a weird issue that occurred only in one of the computed properties I defined in an `11data.js` directory data file. That property depended on another computed property. I have a few such dependencies, but in this particular one, the function received an empty data property and thus it would fail to compute a transormation of another computed property and the build would fail.

After a bit of debugging, I discovered that it occurs when the permalink is set dynamically and reading the debug output, I saw that there is a "ghost" render of the template:
```
Eleventy:Template Rendering permalink for './test/index.njk': (((11ty(((permalink)))11ty))) becomes '(((11ty(((permalink)))11ty)))'
```
I ended up creating (what I think is) a minimal test case project where the same problem occurs. It's available here:
https://github.com/souserge/11ty-permalink-issue-example

The issue occurs in the `test/test.11tydata.js` file. The relevant bit is pasted below:
```js
// pageData comes from pagination in the nunjucks template (test/index.njk)
module.exports = {
  eleventyComputed: {
    permalink: (data) => `/${data.pageData.name}/`,
    nestedDoubleName: (data) => ({ name: data.pageData.name + data.pageData.name }),
    computedProp: (data) => {
      const { nestedDoubleName } = data;
      return nestedDoubleName.name.toUpperCase(); // Error occurs here
    }
  }
}
```

### Reproduction steps

1. [Clone the project](https://github.com/souserge/11ty-permalink-issue-example)
2. Install dependencies
3. Run `DEBUG="Eleventy*" npx @11ty/eleventy`

### Expected behavior

The page isn't rendered without the data being available for computed properties

### Reproduction URL

https://github.com/souserge/11ty-permalink-issue-example

### Screenshots

![image](https://user-images.githubusercontent.com/15277406/222429577-3a77c752-9a2d-4a02-8fb0-802485bd8f17.png)

**Comment by AleksandrHovhannisyan** [comment]
Could this be related to https://github.com/11ty/eleventy/issues/2240?

**Comment by souserge** [comment]
@AleksandrHovhannisyan yes, I think that's what happens here too.

**Comment by Anemolo** [comment]
Stumbuled into this issue after updating to 2.0.1. This used to work on "2.0.0-canary.8", so I'm staying on that version. In my case, I've noticed it happens when using pagination + Permalink:

`_data/renders.js`
```
module.exports = [
	{
		title: "one",
	},
	{
		title: "two",
	},
	{
		title: "three",
	}
]
```

And then `content/key.njk` won't work in this case

```
---
key: My Key
pagination:
  data: renders
  alias: issueData
  size: 1
permalink: "/key/{{issueData.title}}"
eleventyComputed:
  key: "This Is {{key}}"
---

{{key}}

```
But it will work with this setup `content/key.njk``:
```
---
key: My Key
eleventyComputed:
  key: "This Is {{key}}"
---
{{key}}
```

**Comment by alifeee** [comment]
Hi all, this is specifically a problem with nunjucks. If you change your templates to liquid, md, handlebars, etc... it will work properly and `eleventyComputed` values will be computed

more info -> #3120

**Comment by AleksandrHovhannisyan** [comment]
@alifeee OP's example is a JavaScript data file, as are the examples in https://github.com/11ty/eleventy/issues/2240. So I think it's an issue in the data pipeline and not necessarily with the templating languages themselves.

**Comment by alifeee** [comment]
> @alifeee OP's example is a JavaScript data file, as are the examples in https://github.com/11ty/eleventy/issues/2240. So I think it's an issue in the data pipeline and not necessarily with the templating languages themselves.

Ah, you're right. I was perhaps a bit brash thinking "pagination" and "eleventyComputed" would be the same issue as mine.

for this case, the error is raised when eleventy tries to figure out what variables you use in your function ([here](https://github.com/11ty/eleventy/blob/05c62633cc3db0729b7565d5705f76fff1540afb/src/ComputedData.js#L77)), so it does some weird proxy stuff ([here](https://github.com/11ty/eleventy/blob/05c62633cc3db0729b7565d5705f76fff1540afb/src/ComputedDataProxy.js#L123)) which fails because as you say, on the first pass the `data` object is empty. I hope this can be figured out. Thanks.

**Comment by zachleat** [comment]
Let’s group up over at #2240!

For the Nunjucks issue, note that #3120 was merged and shipping with 3.0.0-alpha.11. Thank you!
